projects/rocprofiler-sdk/source/lib/rocprofiler-sdk/pc_sampling/ioctl/ioctl_adapter.cpp

```diff
@@ -517,7 +553,8 @@ ioctl_query_pcs_configs(const rocprofiler_agent_t* agent, rocp_pcs_cfgs_vec_t& r
             // This should never happened, unless the KFD is broken.
             continue;
         }
-        rocp_configs.emplace_back(rocp_cfg);
+        if(is_pc_sampling_firmware_version_correct(agent, get_rocp_pcs_method_from_kfd(ioctl_cfg.method)))
```

Comment:
If you decide to change the is_pc_sampling_firmware_version_correct to return status code, then we could do something like: 

```suggestion
        if(is_pc_sampling_firmware_version_correct(agent, get_rocp_pcs_method_from_kfd(ioctl_cfg.method)) != ROCPROFILER_STATUS_SUCCESS)
```